src/main/java/io/vertx/core/impl/VertxImpl.java

```diff
@@ -86,8 +86,12 @@
   private static final int NETTY_IO_RATIO = Integer.getInteger(NETTY_IO_RATIO_PROPERTY_NAME, 50);
 
   static {
-    // Netty resource leak detection has a performance overhead and we do not need it in Vert.x
-    ResourceLeakDetector.setLevel(ResourceLeakDetector.Level.DISABLED);
+    // Disable Netty's resource leak detection to reduce the performance overhead if not set by user
+    if (System.getProperty("io.netty.leakDetection.level") != null ||
```

Comment:
why 2 distinct property names ?

Reply:
The second property name is a legacy one still used by some applications. I can remove it if you want but I preferred to add it for the sake of completeness, as migrating away from old system properties is not always trivial.

Reply:
can you add a comment about it ?

Reply:
Sure thing 👍

Reply:
Done @vietj